Support sysusers catalog

RIC06X opened the pull request:
### Description

Sysusers contains the mapping between DB users and logins, and to DB roles. Currently Babelfish does not support sysusers catalog. With this commit, we support sys.sysusers catalog.

### Issues Resolved

sysusers catalog is supported in Babelfish
BABEL-2974

### Test Scenarios Covered ###
* **Use case based -**
Test sysusers catalog usage in view
Test sysusers catalog usage in procedure
Test sysusers catalog usage in function
Test sysusers columns for 'dbo', 'guest', sys', 'public', and 'INFORMATION_SCHEMA'. 
Test 'grant/revoke connect' on users.

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

RIC06X committed:
[BABEL-2974] Support sysusers catalog

We add support to sys.sysusers catalog which is not in Babelfish previously.

Task: BABEL-2974
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

xhfanhe commented on contrib/babelfishpg_tsql/sql/ownership.sql:
```diff
@@ -392,6 +392,51 @@ FROM (VALUES ('public', 'R'), ('sys', 'S'), ('INFORMATION_SCHEMA', 'S')) as dumm
+  WHEN Dbp.name = 'guest' AND Ext.user_can_connect = 1 THEN 1
+  WHEN Dbp.type_desc = 'WINDOWS_USER' OR Dbp.name = 'dbo' THEN 1
+  ELSE 0
+END AS hasdbaccess,
```
Why can't we fully rely on user_can_connect?

xhfanhe commented on test/JDBC/expected/sys_sysusers_dep-vu-prepare.out:
```diff
@@ -0,0 +1,35 @@
+BEGIN
+RETURN (SELECT COUNT(*) FROM sys.sysusers WHERE name LIKE '%sysusers_dep_vu_prepare_%')
+END
+GO
```
Please add tests for
1. Users without login
2. Roles
3. guest
4. dbo
5. Users without access

RIC06X replied:
Tests added except first one. Currently babelfish does not support create users without login

forestkeeper commented on test/JDBC/upgrade/latest/schedule:
```diff
@@ -421,3 +421,4 @@ BABEL-3474
 dateadd_internal_df
 datetime2fromparts-after-15-2
 timefromparts
+sys_sysusers_dep
```
What's the result of add sys_sysusers_dep into other mvu schedule ?

RIC06X replied:
I think we are only introducing this catalog in latest version, so I did not add to other upgrade schedule

RIC06X committed:
[BABEL-2974] Fix sys.sysuers' hasdbaccess and islogin column

This commit fixed sys.sysusers hasdbaccess column and islogin column,
modified test cases that will verify essential columns in sysusers for
'dbo', 'guest', sys', 'public', and 'INFORMATION_SCHEMA'. This commit
also added more test cases on 'grant/revoke connect' on users.

Task: BABEL-2974
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

RIC06X committed:
[BABEL-2974] Fix upgrade schedule file

restored upgrade task orderby

Task: BABEL-2974
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

RIC06X committed:
[BABEL-2974] Change test dbname

This commit changed test dbname to sysusersdb and removed 'sys',
'public' and 'INFORMATION_SCHEMA' row from some test cases.

Task: BABEL-2974
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

forestkeeper reviewed (approved):
LGTM
